### PR TITLE
[Release] Complete Linux/SQL & Grep/JDBC Project

### PR DESCRIPTION
**Pull request by Mahmoud-hub**

### CODE REVIEW

File: core_java/grep/out/grep_2022-02-22_21:48:26.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+    Is father, mother, Tybalt, Romeo, Juliet,
```
Comment (emhuskic) [inline core_java/grep/out/grep_2022-02-22_21:48:26.txt]: This file is an output file and should not be contained in this PR.

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
+package ca.jrvs.apps.grep;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.apache.log4j.BasicConfigurator;
+import java.io.*;
```
Comment (emhuskic) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java:7]: It is better to have individual imports, so we can know what packages are we using.

File: core_java/grep/src/main/java/ca/jrvs/apps/practice/RegexExcImp.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        }
+    }
+
+    public static void main(String args[]){
+        RegexExcImp rg=new RegexExcImp();
+        System.out.println(rg.matchJpeg("abc.jpg"));
+        System.out.println(rg.matchIp("192.168.0.1"));
+        System.out.println(rg.isEmptyLine(""));
+    }
+}
```
Comment (emhuskic) [inline core_java/grep/src/main/java/ca/jrvs/apps/practice/RegexExcImp.java]: Every file should have an empty line at its' end.

File: core_java/grep/src/main/java/ca/jrvs/apps/practice/RegexExcImp.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
+package ca.jrvs.apps.practice;
+
+public class RegexExcImp implements RegexExc{
+
+    public boolean matchJpeg(String filename){
+        if ( filename.toLowerCase().matches("[^\\s]+(\\.(jpg|jpeg)$)")) {
```
Comment (emhuskic) [inline core_java/grep/src/main/java/ca/jrvs/apps/practice/RegexExcImp.java]: There is no need for extra spaces after "(" bracket in Java.

File: core_java/jdbc/sql/product.sql [hunk 5]
```diff
@@ -0,0 +1,33 @@
+CREATE SEQUENCE hp_product_seq start with 10000;
```
Comment (emhuskic) [inline core_java/jdbc/sql/product.sql]: You do not have to create a sequence. You can just use `AUTO_INCREMENT` for integers.

File: core_java/jdbc/sql/salesperson.sql [hunk 6]
```diff
@@ -0,0 +1,71 @@
+CREATE SEQUENCE hp_salesperson_seq start with 10000;
+
+
+CREATE TABLE salesperson (
+  salesperson_id bigint NOT NULL DEFAULT nextval('hp_salesperson_seq'),
+  first_name varchar(50) DEFAULT NULL,
```
Comment (emhuskic) [inline core_java/jdbc/sql/salesperson.sql]: There is no need to put default values to NULL explicitly.

File: core_java/jdbc/sql/product.sql (hunk 5 above)
Comment (Mahmoud-hub) [reply core_java/jdbc/sql/product.sql]: I attempted to implement AUTO_INCREMENT, but found that it is not supported with PostgreSQL. Instead I changed the product_id to be a SERIAL type to allow it to increment with every entry without the use of a sequence.

**Pull request merged**